src/Token/LcobucciFactory.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is part of the Mercure Component project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Symfony\Component\Mercure\Token;
+
+use Lcobucci\JWT\Configuration;
+use Lcobucci\JWT\Signer;
+use Lcobucci\JWT\Signer\Key;
+use Symfony\Component\Mercure\Exception\InvalidArgumentException;
+
+if (!class_exists(Key\InMemory::class)) {
+    throw new \LogicException('You cannot use "Symfony\Component\Mercure\Token\LcobucciFactory" as the "lcobucci/jwt" package is not installed. Try running "composer require lcobucci/jwt".');
+}
+
+final class LcobucciFactory implements TokenFactoryInterface
+{
+    /**
+     * @var array<string, class-string<Signer>>
+     */
+    public const SIGN_ALGORITHMS = [
+        'hmac.sha256' => Signer\Hmac\Sha256::class,
+        'hmac.sha384' => Signer\Hmac\Sha384::class,
+        'hmac.sha512' => Signer\Hmac\Sha512::class,
+        'ecdsa.sha256' => Signer\Ecdsa\Sha256::class,
+        'ecdsa.sha384' => Signer\Ecdsa\Sha384::class,
+        'ecdsa.sha512' => Signer\Ecdsa\Sha512::class,
+        'rsa.sha256' => Signer\Rsa\Sha256::class,
+        'rsa.sha384' => Signer\Rsa\Sha384::class,
+        'rsa.sha512' => Signer\Rsa\Sha512::class,
+    ];
+
+    private $configurations;
+
+    public function __construct(string $secret, string $algorithm = 'hmac.sha256')
+    {
+        if (!\array_key_exists($algorithm, self::SIGN_ALGORITHMS)) {
+            throw InvalidArgumentException::forInvalidAlgorithm($algorithm, \array_keys(self::SIGN_ALGORITHMS));
+        }
+
+        $signerClass = self::SIGN_ALGORITHMS[$algorithm];
+        $this->configurations = Configuration::forSymmetricSigner(
+            new $signerClass(),
+            Key\InMemory::plainText($secret)
+        );
+    }
+
+    public function create(array $publish, array $subscribe): string
+    {
+        return $this->configurations
+            ->builder()
+            ->withClaim('mercure', [
```

Comment:
Shouldn't we provide an extension point to add extra claims (especially `exp`)? Maybe can we add a new `$claims` parameter?

Reply:
added 👍